### PR TITLE
RFAI-02: IntentEnvelopeV1, IChatClient adapter, and schema spike

### LINKED ISSUE #974: RFAI-02: IntentEnvelopeV1, IChatClient adapter, and schema spike
## Context

Week 2 from `taskdeck-12-week-roadmap-v4.md`.

Parent: #972
Depends on: #973

This issue creates the versioned intent spine without rewriting the existing capture/chat behavior.

## Scope

- Add versioned domain/application records for `IntentEnvelopeV1`, source references, source blocks, source spans, intent candidates, evidence links, and `TaskdeckProposalBatch`.
- Make Capture and Chat emit envelopes while preserving current parser/planner behavior underneath.
- Spike `Microsoft.Extensions.AI` `IChatClient` behind the existing `ILlmProvider` abstraction for Mock/OpenAI/Gemini, with package versions pinned only after compatibility is measured.
- Spike `System.Text.Json.Schema.JsonSchemaExporter` under .NET 8 for `TaskdeckProposalBatch`.
- If schema exporter compatibility is poor, create a handwritten JSON Schema with snapshot coverage against the C# record shape.
- Add provider-specific strict-schema transforms for OpenAI/Gemini/Ollama smoke paths.

## Acceptance Criteria

- [ ] Capture and Chat both produce a versioned `IntentEnvelopeV1` without changing user-visible behavior.
- [ ] Envelope/source-block/source-span/evidence-link records are stable and test-covered.
- [ ] `IChatClient` spike documents .NET 8 compatibility, package choices, and whether adoption should proceed.
- [ ] Schema generation spike records a clear decision: exporter path or handwritten fallback.
- [ ] Ten-case smoke coverage round-trips `TaskdeckProposalBatch` shape through Mock plus configured provider stubs.
- [ ] `docs/TESTING_GUIDE.md` records the spike outcome and follow-up command path.

## Notes

This is product architecture, not an agent-framework adoption issue. Do not introduce Semantic Kernel or Microsoft Agent Framework here.

## Suggested Verification

- Targeted backend unit tests for envelope/schema records
- Provider smoke tests for Mock/OpenAI/Gemini stubs
- `dotnet build backend/Taskdeck.sln -c Release`

### PR DESCRIPTION
**Pull request by Chris0Jeky**
## Summary

Implements the versioned intent spine for the review-first AI pipeline (#974):

- **6 domain entities**: `IntentEnvelopeV1`, `SourceBlock`, `SourceSpan`, `IntentCandidate`, `EvidenceLink`, `TaskdeckProposalBatch` -- all following existing Entity pattern with private setters, validation via DomainException, and proper lifecycle state machines
- **Application interface**: `IIntentEnvelopeFactory` for creating envelopes from Capture and Chat input as a PARALLEL path (no existing behavior modified)
- **IChatClient spike**: Documented in `docs/spikes/SPIKE_974_ICHATCLIENT.md` -- `Microsoft.Extensions.AI` targets netstandard2.0 and is fully compatible with .NET 8. Thin adapter behind `ILlmProvider` is feasible; deferred to follow-up
- **Schema spike**: `System.Text.Json.Schema.JsonSchemaExporter` requires .NET 9+; handwritten JSON schema at `backend/src/Taskdeck.Application/Schemas/proposal-batch.v1.schema.json`
- **114 new tests**: Unit tests for all 6 entities + 10 schema round-trip smoke tests

## Architecture

```
IntentEnvelopeV1 (spine)
  |-- SourceBlock[] (text blocks with position)
  |     |-- SourceSpan[] (character-level ranges)
  |
  |-- IntentCandidate[] (extracted intents with confidence)
  |     |-- EvidenceLink[] --> SourceSpan (evidence mapping)
  |
  |-- TaskdeckProposalBatch[] (grouped proposals)
```

Lifecycle: Created -> Extracting (first intent added) -> Processed (marked done)

## Key Design Decisions

- Domain entities are pure (no Infrastructure dependencies) -- GP-01 compliant
- Review-first safety preserved -- GP-06: envelopes produce proposals, never direct mutations
- IIntentEnvelopeFactory is a PARALLEL path alongside existing Capture/Chat pipelines
- No Semantic Kernel or Microsoft Agent Framework introduced (per issue requirements)
- TaskdeckProposalBatch stores proposal IDs as flat list to avoid tight coupling with existing AutomationProposal model

## Test Plan

- [x] All 6 domain entities have constructor validation tests
- [x] State machine transitions tested (valid + invalid)
- [x] Boundary value testing for confidence (0.0, 0.5, 1.0), lengths, offsets
- [x] Full lifecycle integration test (envelope -> blocks -> intents -> evidence -> batch -> processed)
- [x] 10 schema round-trip smoke tests covering all operation types, risk levels, complex payloads, UUID preservation
- [x] `dotnet build backend/Taskdeck.sln -c Release` -- 0 errors
- [x] `dotnet test` domain tests -- 1076 total (114 new), all pass

Closes #974

### CODE REVIEW

**Comment by Chris0Jeky** [comment]
## Adversarial Self-Review

### Issues Found and Fixed

**1. BUG: `MarkFailed` accepts but ignores `reason` parameter (IntentEnvelopeV1.cs:156)**
The method signature accepts `string? reason` but never stores it. There is no `FailureReason` property on the envelope. This is either a missing property or a misleading parameter.

**Severity**: Medium -- the reason parameter gives callers a false sense that the failure context is recorded.

**Fix**: Add a `FailureReason` property and store the reason, or remove the parameter. I will add the property.

**2. BUG: `IntentCandidate.AddEvidenceLink` does not validate the link belongs to this candidate**
`AddEvidenceLink(EvidenceLink link)` accepts any link without checking that `link.IntentCandidateId == this.Id`. A caller could accidentally add a link belonging to a different candidate.

**Severity**: Low-Medium -- defensive validation gap.

**Fix**: Add validation check.

**3. Missing test: `SourceBlock.AddSpan` does not validate `startOffset` against content bounds**
`SourceBlock.AddSpan` validates `endOffset > Content.Length` but does NOT validate `startOffset`. A span with `startOffset=100` in a 5-character block would be accepted. The SourceSpan constructor only checks offsets against each other, not against block content.

**Severity**: Low -- semantically wrong but no runtime crash.

**Fix**: Add `startOffset` validation in `AddSpan`.

**4. Architecture: No duplicate position check in `IntentEnvelopeV1.AddSourceBlock`**
Multiple blocks can be added with the same `position` value. This could cause ambiguity in ordering.

**Severity**: Very Low -- the position is informational and the list order is the actual order. No fix needed, just noting for future consideration.

**5. Missing test: `SourceSpan` allows `startOffset=0, endOffset=1` with snippet text longer than the span**
The snippet text length is not validated against `EndOffset - StartOffset`. A 100-char snippet for a 1-char span is semantically wrong.

**Severity**: Very Low -- by design, the snippet is stored independently for auditability. No fix needed.

**6. EF Core: `List<Guid> _proposalIds` in `TaskdeckProposalBatch` may need special EF mapping**
EF Core cannot natively map `List<Guid>` without a value converter or owned type. This is acceptable for now since no EF mapping exists yet (that's a future Infrastructure concern), but should be noted.

**Severity**: None for this PR (domain-only) -- Infrastructure will handle this.

### Applying fixes for items 1, 2, 3

**Comment by Chris0Jeky** [comment]
## Adversarial Review (Round 2)

### Bot findings addressed (all fixed and pushed)

**1. Codex P1 + Gemini medium -- IntentEnvelopeV1.cs:123-128: Status transition before candidate validation**
Fixed. `AddIntentCandidate` now constructs the `IntentCandidate` *before* transitioning status to `Extracting`. If the constructor throws (bad label, confidence, rank, etc.), the envelope remains in `Created` with no half-applied state. Added a regression test (`AddIntentCandidate_ShouldNotTransitionStatus_WhenCandidateValidationFails`).

**2. Gemini medium -- SourceSpan.cs:42: SnippetText length vs span range**
Fixed. The `SourceSpan` constructor now validates that `snippetText.Length == endOffset - startOffset`. Without this, a 5-character snippet could be assigned to a 100-character span range, producing inconsistent evidence data. Added regression test. Also fixed the existing `Constructor_ShouldAcceptBoundarySnippetLength` test which used mismatched values (2000-char snippet, 100-char range) that would have broken.

**3. Gemini medium -- SourceBlock.cs:81: SnippetText must match block content verbatim**
Fixed. `SourceBlock.AddSpan` now verifies that `snippetText == Content.Substring(startOffset, endOffset - startOffset)`. Without this, fabricated snippet text could be attached as "evidence" even when it doesn't match the actual source content -- a data integrity hole that undermines the traceability guarantees. Added regression test.

**4. Codex P1 -- proposal-batch.v1.schema.json:68: Nullable optional fields**
Fixed. Changed `sourceIntentLabel` and `targetId` from `"type": "string"` to `"type": ["string", "null"]`. C# serializes `string?` properties as `null` (not omitted) by default with `System.Text.Json`, so the schema must accept null for optional fields. The round-trip smoke tests (Case01, Case05) emit null for these fields.

**5. Gemini medium -- IntentEnvelopeV1.cs:162: Unused `reason` parameter in `MarkFailed`**
Already addressed in prior self-review commit `382f061c` -- `FailureReason` is stored from the `reason` parameter. No further action needed.

### Additional findings from adversarial review

**A. No `FailureReason` length cap (low severity)**
`MarkFailed(reason)` trims whitespace but does not validate length. A caller could store an arbitrarily long failure reason. Suggest capping at e.g. 2000 characters for consistency with other string limits. Not blocking.

**B. Duplicate positions allowed in `AddSourceBlock` (informational)**
`AddSourceBlock` does not enforce unique `position` values. Two blocks with `Position = 0` can coexist. If position is meant to define ordering, this could lead to ambiguous ordering. If duplicates are intentional (e.g., for replacement scenarios), this is fine. Worth documenting the intent.

**C. `Discard()` is idempotent on `Discarded` status (informational)**
`TaskdeckProposalBatch.Discard()` allows transitioning from `Discarded` to `Discarded`. This is harmless but technically a silent no-op state transition. Noted for awareness.

**D. No duplicate `EvidenceLink` guard by `SourceSpanId` (informational)**
`IntentCandidate.AddEvidenceLink` does not reject adding two links pointing to the same `SourceSpanId`. Multiple links to the same span with different rationale/relevance could be intentional, so this may be by design.

### Verification

- `dotnet build backend/Taskdeck.sln -c Release` -- 0 errors, 12 warnings (pre-existing, unrelated)
- All 122 matching domain tests pass (IntentEnvelope, SourceSpan, SourceBlock, IntentCandidate, EvidenceLink, ProposalBatch)
- Architecture tests pass (no layer boundary violations)
- No Golden Principle violations found (GP-01 layer boundaries, GP-06 review-first)

### Commits pushed

1. `9d1733f9` Move status transition after candidate validation in AddIntentCandidate
2. `a03df3a4` Validate SnippetText length matches span range in SourceSpan constructor
3. `67336acf` Validate snippet text matches block content in SourceBlock.AddSpan
4. `37077528` Allow nullable optional fields in proposal batch schema
5. `51dcadd7` Add tests for adversarial review round 2 fixes

File: backend/src/Taskdeck.Domain/Entities/IntentEnvelopeV1.cs [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    public void MarkProcessed()
+    {
+        if (Status != EnvelopeStatus.Extracting)
+            throw new DomainException(ErrorCodes.InvalidOperation,
+                $"Cannot mark as processed when envelope is in status {Status}");
+        if (_intentCandidates.Count == 0)
+            throw new DomainException(ErrorCodes.ValidationError,
+                "Cannot mark as processed without at least one intent candidate");
+
+        Status = EnvelopeStatus.Processed;
```
[the remark this answers is not in the document]
Comment (Chris0Jeky) [reply backend/src/Taskdeck.Domain/Entities/IntentEnvelopeV1.cs:165]: Fixed in `7d7aa28f`. `MarkProcessed` now requires at least one proposal batch and rejects processing while any associated batch is not `Sealed`, so no-batch and draft-batch partial finalization paths are blocked. Added regressions in `IntentEnvelopeV1Tests`: `MarkProcessed_ShouldRejectWithoutProposalBatch` and `MarkProcessed_ShouldRejectWhenAnyBatchIsDraft`.

Verification:
- `dotnet test backend/tests/Taskdeck.Domain.Tests/Taskdeck.Domain.Tests.csproj -c Release --filter "FullyQualifiedName~IntentEnvelope|FullyQualifiedName~SourceSpan|FullyQualifiedName~SourceBlock|FullyQualifiedName~IntentCandidate|FullyQualifiedName~EvidenceLink|FullyQualifiedName~ProposalBatch"` -- 126 passed
- `dotnet test backend/tests/Taskdeck.Domain.Tests/Taskdeck.Domain.Tests.csproj -c Release` -- 1088 passed
- `dotnet test backend/tests/Taskdeck.Architecture.Tests/Taskdeck.Architecture.Tests.csproj -c Release` -- 16 passed, 4 skipped

File: backend/src/Taskdeck.Domain/Entities/IntentEnvelopeV1.cs [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+    {
+        if (Status == EnvelopeStatus.Processed || Status == EnvelopeStatus.Failed)
+            throw new DomainException(ErrorCodes.InvalidOperation,
+                $"Cannot add intent candidates when envelope is in status {Status}");
+
+        if (Status == EnvelopeStatus.Created)
+            Status = EnvelopeStatus.Extracting;
+
+        var candidate = new IntentCandidate(Id, label, confidence, rank, actionType);
+        _intentCandidates.Add(candidate);
```
[the remark this answers is not in the document]
Comment (Chris0Jeky) [reply backend/src/Taskdeck.Domain/Entities/IntentEnvelopeV1.cs:130]: Current code is fixed. `AddIntentCandidate` constructs and validates `IntentCandidate` before changing `Status` to `Extracting`, and `AddIntentCandidate_ShouldNotTransitionStatus_WhenCandidateValidationFails` covers the partial-write regression. Verified in the focused/domain test runs listed on the latest PR comment thread.

File: backend/src/Taskdeck.Domain/Entities/IntentEnvelopeV1.cs [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+        string? actionType = null)
+    {
+        if (Status == EnvelopeStatus.Processed || Status == EnvelopeStatus.Failed)
+            throw new DomainException(ErrorCodes.InvalidOperation,
+                $"Cannot add intent candidates when envelope is in status {Status}");
+
+        if (Status == EnvelopeStatus.Created)
+            Status = EnvelopeStatus.Extracting;
+
+        var candidate = new IntentCandidate(Id, label, confidence, rank, actionType);
```
[the remark this answers is not in the document]
Comment (Chris0Jeky) [reply backend/src/Taskdeck.Domain/Entities/IntentEnvelopeV1.cs]: Current code is fixed. `AddIntentCandidate` now validates by constructing the candidate before transitioning the envelope status, so constructor failures leave the envelope in `Created`. Covered by `AddIntentCandidate_ShouldNotTransitionStatus_WhenCandidateValidationFails`.

File: backend/src/Taskdeck.Domain/Entities/IntentEnvelopeV1.cs [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+                $"Cannot mark as processed when envelope is in status {Status}");
+        if (_intentCandidates.Count == 0)
+            throw new DomainException(ErrorCodes.ValidationError,
+                "Cannot mark as processed without at least one intent candidate");
+
+        Status = EnvelopeStatus.Processed;
+        Touch();
+    }
+
+    public void MarkFailed(string? reason = null)
```
[the remark this answers is not in the document]
Comment (Chris0Jeky) [reply backend/src/Taskdeck.Domain/Entities/IntentEnvelopeV1.cs:174]: Current code is fixed. `IntentEnvelopeV1` now has `FailureReason`, and `MarkFailed` stores the trimmed reason or `null` for whitespace. Covered by `MarkFailed_ShouldTransitionToFailed`, `MarkFailed_ShouldTrimReason`, and `MarkFailed_ShouldStoreNullReason_WhenWhitespace`.

File: backend/src/Taskdeck.Domain/Entities/SourceSpan.cs [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        string snippetText)
+    {
+        if (sourceBlockId == Guid.Empty)
+            throw new DomainException(ErrorCodes.ValidationError, "SourceBlockId cannot be empty");
+        if (startOffset < 0)
+            throw new DomainException(ErrorCodes.ValidationError, "StartOffset must be non-negative");
+        if (endOffset < 0)
+            throw new DomainException(ErrorCodes.ValidationError, "EndOffset must be non-negative");
+        if (endOffset <= startOffset)
+            throw new DomainException(ErrorCodes.ValidationError, "EndOffset must be greater than StartOffset");
```
[the remark this answers is not in the document]
Comment (Chris0Jeky) [reply backend/src/Taskdeck.Domain/Entities/SourceSpan.cs:46]: Current code is fixed. `SourceSpan` validates `snippetText.Length == endOffset - startOffset`, and the focused RFAI-02 domain test filter passed with the corresponding `SourceSpan` coverage.

File: backend/src/Taskdeck.Domain/Entities/SourceBlock.cs [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+        SourceReferenceId = string.IsNullOrWhiteSpace(sourceReferenceId) ? null : sourceReferenceId.Trim();
+    }
+
+    public SourceSpan AddSpan(int startOffset, int endOffset, string snippetText)
+    {
+        if (endOffset > Content.Length)
+            throw new DomainException(ErrorCodes.ValidationError,
+                $"EndOffset ({endOffset}) exceeds block content length ({Content.Length})");
+
+        var span = new SourceSpan(Id, startOffset, endOffset, snippetText);
```
[the remark this answers is not in the document]
Comment (Chris0Jeky) [reply backend/src/Taskdeck.Domain/Entities/SourceBlock.cs]: Current code is fixed. `SourceBlock.AddSpan` computes the expected substring after bounds validation and rejects snippet text that does not match the block content at the specified offsets. Covered by the focused/domain test runs.

File: backend/src/Taskdeck.Application/Schemas/proposal-batch.v1.schema.json [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+            "Low",
+            "Medium",
+            "High",
+            "Critical"
+          ],
+          "description": "Risk classification for the proposal."
+        },
+        "sourceIntentLabel": {
+          "type": "string",
+          "maxLength": 500,
```
[the remark this answers is not in the document]
Comment (Chris0Jeky) [reply backend/src/Taskdeck.Application/Schemas/proposal-batch.v1.schema.json:68]: Current schema is fixed. `sourceIntentLabel` and `targetId` are typed as `["string", "null"]`, matching the serializer behavior in the round-trip tests where optional fields can be present as `null`.

File: backend/src/Taskdeck.Domain/Entities/SourceBlock.cs [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+    public SourceSpan AddSpan(int startOffset, int endOffset, string snippetText)
+    {
+        if (startOffset > Content.Length)
+            throw new DomainException(ErrorCodes.ValidationError,
+                $"StartOffset ({startOffset}) exceeds block content length ({Content.Length})");
+        if (endOffset > Content.Length)
+            throw new DomainException(ErrorCodes.ValidationError,
+                $"EndOffset ({endOffset}) exceeds block content length ({Content.Length})");
+
+        var expectedSnippet = Content.Substring(startOffset, endOffset - startOffset);
```
[the remark this answers is not in the document]
Comment (Chris0Jeky) [reply backend/src/Taskdeck.Domain/Entities/SourceBlock.cs:87]: Current code is fixed. `SourceBlock.AddSpan` now validates negative/misordered/out-of-bounds offsets before calling `Substring`, so malformed extractor offsets produce `DomainException` instead of `ArgumentOutOfRangeException`. Covered by the focused/domain test runs.

**Comment by Chris0Jeky** [comment]
## Review feedback follow-up

Pushed `7d7aa28f` (`Require sealed batches before processing envelopes`).

Fix:
- `IntentEnvelopeV1.MarkProcessed` now requires at least one proposal batch and rejects processing unless every associated batch is `Sealed`.
- Added regressions for no-batch and draft-batch finalization paths.

Earlier schema/span/status review threads were rechecked against current code and replied to with evidence; those fixes are already present on the branch.

Local verification:
- `dotnet test backend/tests/Taskdeck.Domain.Tests/Taskdeck.Domain.Tests.csproj -c Release --filter "FullyQualifiedName~IntentEnvelope|FullyQualifiedName~SourceSpan|FullyQualifiedName~SourceBlock|FullyQualifiedName~IntentCandidate|FullyQualifiedName~EvidenceLink|FullyQualifiedName~ProposalBatch"` -- 126 passed
- `dotnet test backend/tests/Taskdeck.Domain.Tests/Taskdeck.Domain.Tests.csproj -c Release` -- 1088 passed
- `dotnet test backend/tests/Taskdeck.Architecture.Tests/Taskdeck.Architecture.Tests.csproj -c Release` -- 16 passed, 4 skipped

CI is green for all non-skipped PR checks after the push, and GitHub reports `mergeStateStatus: CLEAN`.

File: backend/src/Taskdeck.Domain/Entities/IntentEnvelopeV1.cs [hunk 9]
```diff
@@ -0,0 +1,187 @@
...
+        if (Status != EnvelopeStatus.Extracting)
+            throw new DomainException(ErrorCodes.InvalidOperation,
+                $"Cannot mark as processed when envelope is in status {Status}");
+        if (_intentCandidates.Count == 0)
+            throw new DomainException(ErrorCodes.ValidationError,
+                "Cannot mark as processed without at least one intent candidate");
+        if (_batches.Count == 0)
+            throw new DomainException(ErrorCodes.ValidationError,
+                "Cannot mark as processed without at least one proposal batch");
+        if (_batches.Any(batch => batch.Status != ProposalBatchStatus.Sealed))
```
[the remark this answers is not in the document]
Comment (Chris0Jeky) [reply backend/src/Taskdeck.Domain/Entities/IntentEnvelopeV1.cs]: Fixed in `ecdab217`. `MarkProcessed` now accepts finalized proposal batches (`Sealed`, `Completed`, or `Discarded`) while still rejecting draft batches/no-batch envelopes, so terminal batches cannot strand the envelope in `Extracting`. Added `MarkProcessed_ShouldAllowTerminalProposalBatches`. Verification: `dotnet test backend/tests/Taskdeck.Domain.Tests/Taskdeck.Domain.Tests.csproj -c Release --filter "FullyQualifiedName~IntentEnvelopeV1" --logger "console;verbosity=minimal"` -- 31 passed.

File: backend/src/Taskdeck.Domain/Entities/IntentCandidate.cs [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+        Confidence = confidence;
+        Rank = rank;
+        ActionType = string.IsNullOrWhiteSpace(actionType) ? null : actionType.Trim();
+    }
+
+    public void AddEvidenceLink(EvidenceLink link)
+    {
+        if (link.IntentCandidateId != Id)
+            throw new DomainException(ErrorCodes.ValidationError,
+                "EvidenceLink does not belong to this IntentCandidate");
```
[the remark this answers is not in the document]
Comment (Chris0Jeky) [reply backend/src/Taskdeck.Domain/Entities/IntentCandidate.cs:83]: Fixed in `ffe98b64`. `IntentCandidate.AddEvidenceLink` now requires the source span object and validates both the link target span id and matching envelope id before accepting evidence, preventing cross-envelope provenance links. Added same-envelope, different-span, and different-envelope regressions. Verification: focused RFAI-02 domain filter passed, 142 total.

File: backend/src/Taskdeck.Domain/Entities/IntentCandidate.cs [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+        string? actionType = null)
+    {
+        if (envelopeId == Guid.Empty)
+            throw new DomainException(ErrorCodes.ValidationError, "EnvelopeId cannot be empty");
+        if (string.IsNullOrWhiteSpace(label))
+            throw new DomainException(ErrorCodes.ValidationError, "Label cannot be empty");
+        if (label.Length > 500)
+            throw new DomainException(ErrorCodes.ValidationError, "Label cannot exceed 500 characters");
+        if (confidence < 0.0 || confidence > 1.0)
+            throw new DomainException(ErrorCodes.ValidationError, "Confidence must be between 0.0 and 1.0");
```
[the remark this answers is not in the document]
Comment (Chris0Jeky) [reply backend/src/Taskdeck.Domain/Entities/IntentCandidate.cs]: Fixed in `ffe98b64`. `IntentCandidate` now rejects non-finite confidence in both construction and update paths. Added NaN/infinity coverage. Verification: focused RFAI-02 domain filter passed, 142 total.

File: backend/src/Taskdeck.Domain/Entities/EvidenceLink.cs [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+        Guid sourceSpanId,
+        double relevance = 1.0,
+        string? rationale = null)
+    {
+        if (intentCandidateId == Guid.Empty)
+            throw new DomainException(ErrorCodes.ValidationError, "IntentCandidateId cannot be empty");
+        if (sourceSpanId == Guid.Empty)
+            throw new DomainException(ErrorCodes.ValidationError, "SourceSpanId cannot be empty");
+        if (relevance < 0.0 || relevance > 1.0)
+            throw new DomainException(ErrorCodes.ValidationError, "Relevance must be between 0.0 and 1.0");
```
[the remark this answers is not in the document]
Comment (Chris0Jeky) [reply backend/src/Taskdeck.Domain/Entities/EvidenceLink.cs]: Fixed in `ffe98b64`. `EvidenceLink` now rejects non-finite relevance in both construction and update paths. Added NaN/infinity coverage. Verification: focused RFAI-02 domain filter passed, 142 total.

File: backend/src/Taskdeck.Domain/Entities/IntentCandidate.cs [hunk 13]
```diff
@@ -0,0 +1,100 @@
...
+        Rank = rank;
+        ActionType = string.IsNullOrWhiteSpace(actionType) ? null : actionType.Trim();
+    }
+
+    public void AddEvidenceLink(EvidenceLink link, SourceSpan sourceSpan)
+    {
+        if (sourceSpan is null)
+            throw new DomainException(ErrorCodes.ValidationError,
+                "SourceSpan is required when adding an EvidenceLink");
+        if (link.IntentCandidateId != Id)
```
[the remark this answers is not in the document]
Comment (Chris0Jeky) [reply backend/src/Taskdeck.Domain/Entities/IntentCandidate.cs:81]: Fixed in `a230aa57`. `IntentCandidate.AddEvidenceLink` now rejects a null `EvidenceLink` with a `DomainException` before dereferencing it. Added `AddEvidenceLink_ShouldRejectNullLink`. Verification: focused `IntentCandidate|ProposalBatchSchemaRoundTrip` domain filter passed, 47 total.

File: backend/tests/Taskdeck.Domain.Tests/Entities/ProposalBatchSchemaRoundTripTests.cs [hunk 14]
```diff
@@ -0,0 +1,282 @@
...
+            });
+    }
+
+    [Fact]
+    public void Case01_MinimalBatch_ShouldRoundTrip()
+    {
+        var batch = CreateMinimalBatch();
+
+        var json = JsonSerializer.Serialize(batch, JsonOptions);
+        var deserialized = JsonSerializer.Deserialize<ProposalBatchDto>(json, JsonOptions);
```
[the remark this answers is not in the document]
Comment (Chris0Jeky) [reply backend/tests/Taskdeck.Domain.Tests/Entities/ProposalBatchSchemaRoundTripTests.cs]: Fixed in `a230aa57`. `ProposalBatchSchemaRoundTripTests` now load the checked-in `proposal-batch.v1.schema.json` and validate every serialized round-trip payload against it, with a negative invalid-risk test proving schema violations fail. Verification: focused `IntentCandidate|ProposalBatchSchemaRoundTrip` domain filter passed, 47 total.

File: backend/src/Taskdeck.Application/Schemas/proposal-batch.v1.schema.json [hunk 15]
```diff
@@ -0,0 +1,117 @@
...
+        }
+      }
+    },
+    "operation": {
+      "type": "object",
+      "additionalProperties": false,
+      "required": [
+        "operationType",
+        "payload"
+      ],
```
[the remark this answers is not in the document]
Comment (Chris0Jeky) [reply backend/src/Taskdeck.Application/Schemas/proposal-batch.v1.schema.json:119]: Fixed in `475c717e`. `proposal-batch.v1.schema.json` now applies an `if`/`then` rule requiring a UUID `targetId` for non-create operations while still allowing create operations to carry null. Added schema regression coverage for non-create operations with null targets. Verified with `dotnet test backend/tests/Taskdeck.Domain.Tests/Taskdeck.Domain.Tests.csproj -c Release --filter ProposalBatchSchemaRoundTripTests --logger "console;verbosity=minimal"` (17 passed).
